MetaMask/metamask-extension
# Fix incorrect balance in signature request header

Fixes: https://github.com/MetaMask/metamask-extension/issues/17541

## Explanation
The network native currency was always being chosen to be displayed here, `useNativeCurrencyAsPrimaryCurrency` should be taken in to consideration so `currentCurrency` can be shown when appropriate.

## Screenshots/Screencaps
<img width="637" alt="Screen Shot 2023-02-22 at 8 22 58 AM" src="https://user-images.githubusercontent.com/8732757/220671198-47730feb-02e7-4d63-afbf-7d52335f49fd.png">
<img width="616" alt="Screen Shot 2023-02-22 at 8 23 11 AM" src="https://user-images.githubusercontent.com/8732757/220671210-66ac884d-0a36-42ff-8c9e-a6bd85bb336b.png">

## Manual Testing Steps
1. Settings -> Advanced -> Show conversion on test networks -> ON
2. Settings -> General -> Primary currency -> Fiat 
3. Navigate to [test dapp](https://metamask.github.io/test-dapp/)
4. Click `Sign` in the `Sign Typed Data V4` section
5. Confirm that `USD` is displayed as the currency type in the header

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
